android/java/org/chromium/chrome/browser/vpn/utils/BraveVpnPrefUtils.java

```diff
@@ -160,8 +160,13 @@ public static void setPurchaseToken(String value) {
         SharedPreferences.Editor sharedPreferencesEditor = mSharedPreferences.edit();
         sharedPreferencesEditor.putString(PREF_BRAVE_VPN_PURCHASE_TOKEN, value);
         sharedPreferencesEditor.apply();
+        String[] purchaseTokenParts = value.split(".");
+        String purchaseToken = (purchaseTokenParts.length == 2
+                                && BraveVpnUtils.isBase64UrlEncoded(purchaseTokenParts[0]))
```

Comment:
you probably still want to check that the second part is b64 encoded?

Reply:
The second part definitely isn't - it has a `-` in it and will fail parsing ☹️

Reply:
update - talked with @deeppandya; his version of the regex handles `_` and `-` which meets our criteria; we're going to revert my changes and give it a go again 😄 Problem I ran into may have been just server side or anomaly